ext/ddtrace_profiling_native_extension/collectors_discrete_dynamic_sampler.c

```diff
@@ -347,29 +331,39 @@ static const rb_data_type_t sampler_typed_data = {
 static VALUE _native_new(VALUE klass) {
   sampler_state *state = ruby_xcalloc(sizeof(sampler_state), 1);
 
-  discrete_dynamic_sampler_init(&state->sampler, "test sampler");
-
   return TypedData_Wrap_Struct(klass, &sampler_typed_data, state);
 }
```

Comment:
Minor: Should we still perhaps call `_init` here (possibly with a dummy time), just to avoid exposing the non-initialized sampler to Ruby code?